client/Wayland/wlfreerdp.c

```diff
@@ -117,14 +137,37 @@ BOOL wl_post_connect(freerdp* instance)
 	 /* put Wayland data in the context here */
 	context->window = window;
 
-	freerdp_channels_post_connect(instance->context->channels, instance);
+	if (freerdp_channels_post_connect(instance->context->channels, instance))
+		return FALSE;
```

Comment:
If I'm not mistaken in error case here are some possible memory leaks (gdi,  window, window->data)